src/backend/cpu/sparse.cpp

```diff
@@ -339,18 +339,52 @@ Array<T> sparseConvertStorageToDense(const SparseArray<T> &in_)
 ////////////////////////////////////////////////////////////////////////////////
 // Common to MKL and Not MKL
```

Comment:
There are 2 code paths: One that uses MKL and one that is used when compiled without MKL support. This section is common to both as it does not depend on MKL.